src/components/Education.js

```diff
@@ -1,34 +1,131 @@
 import styles from './education.module.css';
+import { useState, Fragment } from 'react';
+import { format } from 'date-fns';
 
-function handleClick(e) {}
+export default function Education({ data, onChange, onClick }) {
+  const [isEditing, setIsEditing] = useState(false);
+
+  function handleClick() {
+    isEditing ? setIsEditing(false) : setIsEditing(true);
+  }
+
+  function generateContent(schoolData) {
```

Comment:
Personally, all those generateComponent() functions get's out of hand really quick. It's much simpler to read and edit if this generateContent would be separated to new Component in new File with a proper name and the same logic would be done for a standalone component.

Reply:
so every component (General, Education etc.) should have their own content generator component, f.e. GeneralContentGenerator.js? or do I create one and pass different data to it?

Reply:
If the generated content is repetitive, you could do that with reusing one component, of course. Also <GeneralContent/> and not <GeneralContentGenerator/>